### PR TITLE
BankSignering: implement contract query

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Implement new Contract query on BankSignering page

- Extract reusable hook to convert API contract to "BankSigneringContract"

- Implement different states for BankSignering page

- Pass authentication status from server to avoid hydration errors

## Justify why they are needed

- Complete the feature

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2634** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2634" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2634?utm_source=stack-comment).

File: apps/store/src/components/ConfirmationPage/useSwitchingContracts.ts [hunk 1]
```diff
@@ -20,36 +20,59 @@ export const useSwitchingContracts = ({ shopSessionId }: Params) => {
...
+    externalInsurer: { displayName: string }
+    status: ContractExternalInsuranceCancellationStatus
+    bankSignering?: { approveByDate: string } | null
+  } | null
+}
+
+export const useConvertToBankSigneringContract = (): ((
+  contract: MinimalContract,
+) => BankSigneringContract | undefined) => {
+  const getStatus = useGetStatus()
```
Comment (alebedev) [inline apps/store/src/components/ConfirmationPage/useSwitchingContracts.ts:57]: We have chain of 2 hooks where pure functions could be used.  As I can see, only hook dependency is i18n - can we return translation key and translate it in JSX?

File: apps/store/src/components/InitiateCarCancellationPage/InitiateCarCancellationPage.tsx [hunk 2]
```diff
@@ -30,6 +79,36 @@ export const InitiateCarCancellationPage = (props: Props) => {
...
+        />
+        <Button type="submit">{t('LOGIN_BUTTON_TEXT')}</Button>
+      </form>
+    </Layout>
+  )
+}
+
+const Loading = CardSkeleton
+
+// TODO: Localize these texts
```
Comment (alebedev) [inline apps/store/src/components/InitiateCarCancellationPage/InitiateCarCancellationPage.tsx:99]: Should it be a part of this PR?

File: apps/store/src/graphql/Contract.graphql [hunk 3]
```diff
@@ -0,0 +1,18 @@
+query Contract($id: UUID!) {
+  contract(id: $id) {
+    id
```
Comment (alebedev) [inline apps/store/src/graphql/Contract.graphql:3]: Can we use some fragment inside?  Would be easier to type data in the code

**Review by alebedev — APPROVED** [review]
Great work!  1 comment inside

File: apps/store/src/components/InitiateCarCancellationPage/InitiateCarCancellationPage.tsx (hunk 2 above)
Comment (robinandeer) [reply apps/store/src/components/InitiateCarCancellationPage/InitiateCarCancellationPage.tsx:99]: I think we are still waiting on some input on the design of this page

But perhaps we can have a go at it

**Pull request merged**